### ISSUE TITLE
Nullable collection elements

### ISSUE THREAD

**Issue by travikk** [vojtechhabarta/typescript-generator#304]
Hey there!

I'm using Kotlin to create all my DTOs. In Kotlin, you can do the following:

```kotlin
data class MyDto(
  val arrWithElements: Array<Int>,
  val arrWithNullableElements: Array<Int?>
)
```

The information doesn't exist at runtime unfortunately, as it's wiped away, therefore there's no easy direct way to generate an interface in TS that would look like this:

```typescript
interface MyDto {
  arrWithElements: number[]; // or Array<number>
  arrWithNullableElements: Array<number | null>;
}
```

What I'm proposing then, is the ability to configure a custom annotation in the generator, which if defined on a property that happens to be a collection, that collection would be understood to have nullable elements.

Please let me know what you think

**Comment by vojtechhabarta** [comment]
Hey, thanks for suggestion with examples.

You are right that this could be supported but I am not sure that using an annotation is a good solution. I could have for example type `Array<Array<Int?>>` or `Array<Array<Int>?>`. It would not be possible to produce "right" TypeScript type in these cases.

Currently typescript-generator is not able to represent nullable types so first the internal model needs to be extended. Then I would suggest to try to use Kotlin reflection (https://kotlinlang.org/docs/reference/reflection.html, https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/index.html) to get correct types.

**Comment by travikk** [comment]
@vojtechhabarta good point.

I thought the model already supported nullable types, since there is support for Optional type, no?

**Comment by vojtechhabarta** [comment]
@danielkaneider once we have support in model for `NullableType` this could also be supported.

**Comment by RobbinBaauw** [comment]
@danielkaneider in https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.19.577 this has been released!

**Comment by vojtechhabarta** [comment]
Yes, support for nullability in Kotlin (and also in Java) was released. Initially in [2.19.575](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.19.575) and some bug fixes in [2.19.577](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.19.577) and [2.20.583](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.20.583).
